src/service/repo/signedMessageRepo.ts

```diff
@@ -49,8 +62,17 @@ async function checkIsTargetAllowingAnonDM(targetUserId: string) {
   }
 }
 
-async function sendSignedMessage(channelId: string, message: string, channelType: number) {
-  const payload = {channelId, message, channelType};
+async function sendSignedMessage(
+  channelId: string,
+  message: string,
+  channelType: number,
+  replyMessageId?: string
+) {
+  let payload: SendPayloadType = {channelId, message, channelType};
+  if (replyMessageId) {
+    payload = {...payload, messageType: 'reply', replyMessageId};
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The implementation of `sendSignedMessage` has been updated to handle the case where a `replyMessageId` is provided. The payload is constructed conditionally based on whether `replyMessageId` is present or not. If it is, the `messageType` is set to 'reply' and `replyMessageId` is included in the payload. This is a good change as it allows for the functionality of replying to specific messages in a chat. However, I would suggest initializing the payload object once and then just updating it if necessary, rather than reassigning the entire object. This can be done as follows:

```diff
- let payload: SendPayloadType = {channelId, message, channelType};
- if (replyMessageId) {
-   payload = {...payload, messageType: 'reply', replyMessageId};
- }
+ const payload: SendPayloadType = {channelId, message, channelType};
+ if (replyMessageId) {
+   payload.messageType = 'reply';
+   payload.replyMessageId = replyMessageId;
+ }
```
This way, you avoid creating a new object unnecessarily, which can be more efficient especially when dealing with larger objects.


<!-- This is an auto-generated comment by OSS CodeRabbit -->